pkg/kv/kvserver/rangefeed/budget.go

```diff
@@ -49,6 +50,25 @@ var maxFeedFraction = envutil.EnvOrDefaultFloat64("COCKROACH_RANGEFEED_TOTAL_MEM
 var systemRangeFeedBudget = envutil.EnvOrDefaultInt64("COCKROACH_RANGEFEED_SYSTEM_BUDGET",
 	2*64*1024*1024 /* 128MB */)
 
+// RangefeedBudgetsEnabled is a cluster setting that enables rangefeed memory
+// budgets. This is meant to be an escape hatch to disable budgets if they cause
+// feeds to fail unexpectedly despite nodes have plenty of memory or if bugs in
+// budgeting are discovered and mitigation is required.
+// With budgets disabled, rangefeeds will behave as they did prior to 22.1.
+// Note that disabling this feature will not disable budgets on already running
+// rangefeeds, but will affect feeds that are created after the option was
+// changed. You need to restart all feeds to ensure that the budget is fully
+// disabled.
```

Comment:
I don't think this is true? If I have a running rangefeed, change this setting, and then create another rangefeed on that replica, then the second rangefeed will reuse the existing processor with the existing budget. It only takes effect when all rangefeeds are completely stopped. Also, this should be mentioned in the setting description itself, e.g. `(takes effect after stopping all existing rangefeeds)`, because that's what users see.

I suppose dynamically disabling the budget would be too hard?

Reply:
This is true. I didn't think about multiple feeds on the range. Let's try to make it more dymanic in follow up change as discussed offline.